api/lib/source/polkadotV0-source.js

```diff
@@ -529,6 +532,151 @@ class polkadotAPI {
       }
     )
   }
+
+  async getAllProposals() {
+    const api = await this.getAPI()
+
+    const [
+      blockHeight,
+      totalIssuance,
+      democracyProposals,
+      democracyReferendums,
+      treasuryProposals,
+      councilProposals,
+      councilMembers
+    ] = await Promise.all([
+      this.getBlockHeight(),
+      api.query.balances.totalIssuance(),
+      api.derive.democracy.proposals(),
+      api.derive.democracy.referendums(),
+      api.derive.treasury.proposals(),
+      api.derive.council.proposals(),
+      api.query.council.members()
+    ])
+    const allProposals = democracyProposals
+      .map((proposal) => {
+        return this.reducers.democracyProposalReducer(
+          this.network,
+          proposal,
+          totalIssuance,
+          blockHeight
+        )
+      })
+      .concat(
+        democracyReferendums.map((proposal) => {
+          return this.reducers.democracyReferendumReducer(
+            this.network,
+            proposal,
+            totalIssuance,
+            blockHeight
+          )
+        })
+      )
+      .concat(
+        treasuryProposals.proposals.map((proposal) => {
+          return this.reducers.treasuryProposalReducer(
+            this.network,
+            proposal,
+            councilMembers
+          )
+        })
+      )
+      .concat(
+        councilProposals.map((proposal) => {
+          return this.reducers.councilProposalReducer(
+            this.network,
+            proposal,
+            councilMembers,
+            blockHeight
+          )
+        })
+      )
+
+    return orderBy(allProposals, 'id', 'desc')
+  }
+
+  getProposalById() {}
+
+  getDelegatorVote() {}
+
+  async getTotalActiveAccounts() {
+    const api = await this.getAPI()
+    const accountKeys = await api.query.system.account.keys()
+    const accounts = accountKeys.map((key) => key.args[0].toHuman())
+    return accounts.length || 0
+  }
+
+  async getRecentProposals() {
+    // while we don't have proposals in DB this is the only way
+    const proposals = await this.getAllProposals()
+    // the problem in Substrate is that right now we don't have all creationTimes yet
+    return proposals.sort(
+      (a, b) => new Date(b.creationTime) - new Date(a.creationTime)
+    )
+  }
+
+  async getTopVoters() {
+    // in Substrate we simply return council members
+    const api = await this.getAPI()
+    const councilMembers = await api.query.council.members()
+    const councilMembersWithStake = await Promise.all(
+      councilMembers.map(async (member) => {
+        const account = await api.query.system.account(member)
+        const totalBalance = account.data.free
+        const freeBalance = BigNumber(totalBalance.toString()).minus(
+          account.data.miscFrozen.toString()
+        )
+        return {
+          address: member,
+          stakedBalance: totalBalance - freeBalance
```

Comment:
you are double negating here.
```suggestion
          stakedBalance: Number(account.data.miscFrozen.toString())
```